tests/sentry/utils/kvstore/test_common.py

```diff
@@ -0,0 +1,93 @@
+import itertools
+import pytest
+from dataclasses import dataclass
+from typing import Iterator, Tuple
+
+from sentry.utils.kvstore.abstract import K, KVStorage, V
+
+
+@dataclass
+class Properties:
+    store: KVStorage[K, V]
+    keys: Iterator[K]
+    values: Iterator[V]
+
+    @property
+    def items(self) -> Iterator[Tuple[K, V]]:
+        return zip(self.keys, self.values)
+
+
+@pytest.fixture(params=["bigtable", "cache/default"])
+def properties(request) -> Properties:
+    if request.param == "bigtable":
+        from tests.sentry.utils.kvstore.test_bigtable import create_store, get_credentials
+
+        return Properties(
+            create_store(request, get_credentials()),
+            keys=(f"{i}" for i in itertools.count()),
+            values=(f"{i}".encode("utf-8") for i in itertools.count()),
+        )
+    elif request.param.startswith("cache/"):
+        from sentry.utils.kvstore.cache import CacheKVStorage
+
+        # XXX: Currently only testing against the default cache is supported
+        # because testing against the Redis cache requires complex mocking of
+        # global state in ``sentry.utils.redis``.
```

Comment:
The idealist in me wants to do this "correctly"/comprehensively but the pragmatist in me thinks this is unlikely to uncover anything interesting and won't be worth the investment.